xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/suggest/suggestPages.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+require.config({
+  paths: {
+    'xwiki-selectize': "$xwiki.getSkinFile('uicomponents/suggest/xwiki.selectize.js', true)" +
+      "?v=$escapetool.url($xwiki.version)"
+  }
+});
+
+define('xwiki-suggestPages', ['jquery', 'xwiki-selectize'], function($) {
+  'use strict';
+
+  var pageIcon = $jsontool.serialize($services.icon.getMetaData('page_white'));
+
+  var getSelectizeOptions = function(select) {
+    return {
+      create: true,
+      load: function(text, callback) {
+        loadPages(text).done(function(data) {
+          var pages = [];
+          data.searchResults.forEach(function (element) {
+            var hierarchy = element.hierarchy.items;
+            var label = hierarchy.pop().label;
+            if (element.pageName === 'WebHome') {
+              label = hierarchy.pop().label;
+            }
```

Comment:
That's fine, but I would probably take the label from ``element.title`` and use the `hierarchy`` just to compute the hint.

Reply:
`element.title` contains the display title whereas the `hierarchy` has the plain title for spaces and documents and the name of the entity for other types.
https://github.com/xwiki/xwiki-platform/blob/b4095dbabf797c0cbb176b3625e29ce1a5b27d20/xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/BaseSearchResult.java#L254
https://github.com/xwiki/xwiki-platform/blob/b4095dbabf797c0cbb176b3625e29ce1a5b27d20/xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/BaseSearchResult.java#L293-L298

Reply:
OK.